library-model/library-model-generator/src/test/java/com/uber/nullaway/libmodel/LibraryModelGeneratorTest.java

```diff
@@ -86,6 +93,62 @@ public void nullableUpperBound() throws IOException {
         };
     ImmutableMap<String, Set<Integer>> expectedNullableUpperBounds =
         ImmutableMap.of("NullableUpperBound", ImmutableSet.of(0));
-    runTest("NullableUpperBound.java", lines, ImmutableMap.of(), expectedNullableUpperBounds);
+    runTest(
+        "NullableUpperBound.java",
+        lines,
+        ImmutableMap.of(),
+        expectedNullableUpperBounds,
+        ImmutableSet.of("NullableUpperBound"));
+  }
+
+  @Test
+  public void nullMarkedClasses() throws IOException {
+    String[] lines =
+        new String[] {
+          "import org.jspecify.annotations.NullMarked;",
+          "@NullMarked",
+          "public class NullMarked{",
+          "//Only this class should appear under null marked classes",
+          "}",
+          "class NotNullMarked{",
+          "}"
+        };
+    ImmutableSet<String> expectedNullMarkedClasses = ImmutableSet.of("NullMarked");
+    runTest(
+        "NullableUpperBound.java",
+        lines,
+        ImmutableMap.of(),
+        ImmutableMap.of(),
+        expectedNullMarkedClasses);
+  }
+
+  @Test
+  public void nullableParameters() throws IOException {
+    String[] lines =
+        new String[] {
+          "import org.jspecify.annotations.NullMarked;",
+          "import org.jspecify.annotations.Nullable;",
+          "@NullMarked",
```

Comment:
Maybe add another test `nullableParametersInNullUnmarkedClass()` to make sure we don't get any records for that case?